sys/include/entropy_source/entropy_source_common.h

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup   sys_entropy_source_common Entropy Source Common
+ * @ingroup    sys_entropy_source
+ * @brief      Common definitions and functions for entropy sources
+ *
+ * @{
+ * @file
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ */
+
+#ifndef ENTROPY_SOURCE_COMMON_H
+#define ENTROPY_SOURCE_COMMON_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stddef.h>
+#include <inttypes.h>
+
+/**
+ * @brief   Entropy source error codes.
+ */
+typedef enum {
+    ENTROPY_SOURCE_OK              =  0,   /**< Success */
+    ENTROPY_SOURCE_ERR_INIT        = -1,   /**< Source initialization error */
+    ENTROPY_SOURCE_ERR_CONFIG      = -2,   /**< Source configuration error */
+    ENTROPY_SOURCE_ERR_TEST_REP    = -3,   /**< Repetition count test error */
+    ENTROPY_SOURCE_ERR_TEST_PROP   = -4,   /**< Adaptive proportion test error */
+    ENTROPY_SOURCE_ERR_TEST_BOTH   = -5,   /**< Repetition count and Adaptive proportion
+                                             *  test error */
+    ENTROPY_SOURCE_ERR_COND        = -6,   /**< Conditioning error */
+} entropy_source_error_t;
+
+/**
+ * @brief   Data structure for Repetition Count Test (NIST SP 800-90B 4.4.1).
+ */
+typedef struct {
+    uint8_t old_sample;    /**< Preceding sample to compare for repetition */
+    uint16_t cnt_rep;      /**< Counter to count repetition */
+    uint8_t c_rep;         /**< Cutoff threshold */
+} entropy_source_tests_rep_t;
+
+/**
+ * @brief   Data structure for Adaptive Proportion Test (NIST SP 800-90B 4.4.2).
+ */
+typedef struct {
+    uint8_t old_sample;    /**< Preceding sample to compare for repetition */
+    uint16_t cnt_prop;     /**< Counter to count proportion */
+    uint16_t cnt_window;   /**< Counter to count window size */
+    uint16_t c_prop;       /**< Cutoff threshold */
+} entropy_source_tests_prop_t;
+
+/**
+ * @brief   Scale Min. Entropy to fixed point integer to avoid float. The entropy per sample
+            (8 Byte) of a noise source can likely be smaller than 1 bit.
+ */
+#define ENTROPY_SOURCE_HMIN_SCALE(x)    ((unsigned) (x * (1 << 16)))
+
+/**
+ * @brief   Scale internal fixed point Min. Entropy back to float. This macro is not
+ *          required and only there for convenience.
+ */
+#define ENTROPY_SOURCE_HMIN_SCALE_BACK(x)    (((float)x / (1 << 16)))
+
+/**
+ * @brief   Window size for Adaptive Proportion Test (NIST SP 800-90B 4.4.2).
```

Comment:
These can be added to the compile time config group.

Reply:
Isn't [this](https://github.com/RIOT-OS/RIOT/pull/14324/files#diff-425caf902000cbd39fad527cec594833R15) enough?

Reply:
No. I mean in doxygen. We have a `@config` group http://doc.riot-os.org/group__config.html